### PR TITLE
[ACM-13054] Added sample service-account authentication secret

### PR DESCRIPTION
**Pull request by dislbenn**
# Description

As we prepare for the transition of moving away from OCM offline tokens for discovery, we will begin supporting OCM service accounts for authenticating users to discover clusters from OCM.

## Related Issue

https://issues.redhat.com/browse/ACM-13054

## Changes Made

Added sample service account secret for discovery.

## Screenshots (if applicable)

Add screenshots or GIFs that demonstrate the changes visually, if relevant.

## Checklist

- [x] I have tested the changes locally and they are functioning as expected.
- [ ] I have updated the documentation (if necessary) to reflect the changes.
- [ ] I have added/updated relevant unit tests (if applicable).
- [x] I have ensured that my code follows the project's coding standards.
- [x] I have checked for any potential security issues and addressed them.
- [x] I have added necessary comments to the code, especially in complex or unclear sections.
- [ ] I have rebased my branch on top of the latest main/master branch.

## Additional Notes

Add any additional notes, context, or information that might be helpful for reviewers.

## Reviewers

Tag the appropriate reviewers who should review this pull request. To add reviewers, please add the following line: `/cc @reviewer1 @reviewer2`

## Definition of Done

- [ ] Code is reviewed.
- [ ] Code is tested.
- [ ] Documentation is updated.
- [x] All checks and tests pass.
- [ ] Approved by at least one reviewer.
- [ ] Merged into the main/master branch.

### CODE REVIEW

File: config/samples/ocm-api-secret.yaml [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 apiVersion: v1
 data:
```
Comment (gparvin) [inline config/samples/ocm-api-secret.yaml:4]: A sample that doesn't show the data base64 encoded should use `stringData` like this example: https://kubernetes.io/docs/concepts/configuration/secret/#basic-authentication-secret

File: config/samples/ocm-api-serviceaccount.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# Copyright Contributors to the Open Cluster Management project
+
+apiVersion: v1
+data:
```
Comment (gparvin) [inline config/samples/ocm-api-serviceaccount.yaml:4]: Probably want to show this as `stringData` too

File: config/samples/ocm-api-secret.yaml (hunk 1 above)
Comment (dislbenn) [reply config/samples/ocm-api-secret.yaml]: Updated the field to use `stringData`.

File: config/samples/ocm-api-serviceaccount.yaml (hunk 2 above)
Comment (dislbenn) [reply config/samples/ocm-api-serviceaccount.yaml]: Updated the field to use `stringData`.

**Review by gparvin — APPROVED** [review]

**Pull request merged**